java/java.hints/src/org/netbeans/modules/java/hints/perf/Tiny.java

```diff
@@ -462,36 +485,94 @@ public static ErrorDescription redundantToString(HintContext ctx) {
         suppressWarnings = "UnnecessaryTemporaryOnConversionFromString" 
     )
     public static ErrorDescription unnecessaryTempFromString(HintContext ctx) {
-        TypeMirror resType = ctx.getInfo().getTrees().getTypeMirror(ctx.getPath());
-        if (resType == null) {
-            return null;
+
+        String type;
+        String method;
+
+        // determine if the destination is primitive
+        TypeMirror destType = getDestinationType(ctx, ctx.getPath());
+        TypeMirror srcType = ctx.getInfo().getTrees().getTypeMirror(ctx.getPath());
+
+        if (destType instanceof PrimitiveType && !(srcType instanceof PrimitiveType)) {
```

Comment:
Generally, the javac APIs discourage `instanceof` and encourage use of `.getKind() == <type>` (or probably `.getKind().isPrimitive()` in this case), even though that obviously lacks the implicit null check. The reason is that one implementation class may implement many of these interfaces (applies to all `javax.lang.model.**` and `com.sun.source.**` that have kinds). As a real-world example, the `com.sun.tools.javac.code.Type.ClassType` implements both `DeclaredType` and `ErrorType`, so doing `instanceof DeclaredType` or `instanceof ErrorType` would not work properly. The `.getKind()` returns the correct kind for the given type.

Reply:
good to know -> switched everything to getKind()